### ISSUE TITLE
why gson can parse some string not stand json such as "[a,b,c]" ?

### ISSUE THREAD

**Issue by somewon** [google/gson#763]
``` java
        JsonParser parser = new JsonParser();
        JsonElement el = parser.parse("[a,b,c]");
        JsonArray array = new JsonArray();
        array = el.getAsJsonArray();
```

? why?

**Comment by JakeWharton** [comment]
It's not valid JSON. Try `"[\"a\", \"b\", \"c\"]"` instead.

**Comment by somewon** [comment]
Yes. I know that.  It's not valid JSON. But gson could parse this. 
I'm curious about why gson have this feature.

**Comment by JakeWharton** [comment]
Oh you're asking why it _can_ parse it? For historical reasons Gson is lenient by default and unfortunately there's no sane API to disable it for `JsonParser` that I can see. Even if you create your own `JsonReader` and call `setLenient(false)` Gson will ignore that setting.

Disabling the lenient behavior can be done when using the object binding API (via the `Gson` type), but there's hoops to jump through. Are you able to use the binding API instead of the DOM API?

**Comment by pretizy** [comment]
I haven't reviewed the code, but can you explain why the lenient behavior
affects the json standard format

**Comment by Marcono1234** [comment]
Probably resolved (but not really fixed) by the documentation improvements of #2122.

**Comment by eamonnmcmanus** [comment]
I think those documentation improvements are probably the best we can do given compatibility requirements and the maintenance-mode status of this project.
